src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,218 @@
+                request[(int)MessageFormat.Checksum] = Checksum(request);
+
+                _serialPort.Open();
+                _serialPort.Write(request, 0, request.Length);
+
+                byte[] response = new byte[MessageSize];
+                if ((_serialPort.Read(response, 0, response.Length) == response.Length) && (response[(int)MessageFormat.Checksum] == Checksum(response)))
+                {
+                    concentration = Ratio.FromPartsPerMillion((int)response[(int)MessageFormat.DataHigh] * 256 + (int)response[(int)MessageFormat.DataLow]);
+                    validity = true;
+                }
+            }
+            finally
+            {
+                _serialPort.Close();
+            }
+
+            return (concentration, validity);
+        }
+
+        /// <summary>
+        /// Initiates a zero point calibration.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <returns>true, if the command could be send</returns>
+        public bool ZeroPointCalibration() => SendRequest(CreateRequest(Command.CalibrateZeroPoint));
+
+        /// <summary>
+        /// Initiate a span point calibration.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <param name="span">span value, e.g. 2000[ppm]</param>
+        /// <returns>true, if the command could be send</returns>///
+        public bool SpanPointCalibration(int span)
+        {
+            var request = CreateRequest(Command.CalibrateSpanPoint);
+            // set span in request, c. f. datasheet rev. 1.0, pg. 8 for details
+            request[(int)MessageFormat.DataHigh] = (byte)(span / 256);
+            request[(int)MessageFormat.DataLow] = (byte)(span % 256);
+
+            return SendRequest(request);
+        }
+
+        /// <summary>
+        /// Switches the autmatic baseline correction on and off.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <param name="state">State of automatic correction</param>
+        /// <returns>true, if the command could be send</returns>
+        public bool AutomaticBaselineCorrection(AbmState state)
+        {
+            var request = CreateRequest(Command.AutoCalibrationSwitch);
+            // set on/off state in request, c. f. datasheet rev. 1.0, pg. 8 for details
+            request[(int)MessageFormat.DataHigh] = (byte)state;
+
+            return SendRequest(request);
+        }
+
+        /// <summary>
+        /// Set the sensor detection range.
+        /// The sensor doesn't respond anything, so this is fire and forget
+        /// </summary>
+        /// <param name="detectionRange">Detection range of the sensor</param>
+        /// <returns>true, if the command could be send</returns>
+        public bool SensorDetectionRange(DetectionRange detectionRange)
```

Comment:
You can either use a property here or add an action verb like SetSensorDetectionRange. 
Having a property would a bit better. You can as well return the current value.

Reply:
Same comment as for AutomaticBaselineCorrection command. I would prefer a method over a property.

Reply:
Then please adjust the name with an action verb

Reply:
Done. Method renamed to "SetSensorDectectionRange".
Returning the current setting is not possible as there is no command to read out the current setting from the sensor device.